front/components/home/new/SolutionSection.tsx

```diff
@@ -0,0 +1,151 @@
+    | SolutionSectionContentBlockProps[];
+  assistantBlocks:
+    | SolutionSectionAssistantBlockProps
+    | SolutionSectionAssistantBlockProps[];
+  color?: "pink" | "sky" | "emerald" | "amber";
+  className?: string;
+}
+
+SolutionSection.Block = function SolutionSectionBlock({
+  contentBlocks,
+  assistantBlocks,
+  color = "emerald",
+  className = "",
+}: SolutionSectionBlockProps) {
+  const renderContentBlocks = () => {
+    if (Array.isArray(contentBlocks)) {
+      return contentBlocks.map((block, index) => (
+        <SolutionSection.ContentBlock key={index} {...block} />
+      ));
+    } else {
+      return <SolutionSection.ContentBlock {...contentBlocks} />;
+    }
+  };
+
+  const renderAssistantBlocks = () => {
+    if (Array.isArray(assistantBlocks)) {
+      return assistantBlocks.map((block, index) => (
+        <SolutionSection.AssistantBlock key={index} {...block} />
+      ));
+    } else {
+      return <SolutionSection.AssistantBlock {...assistantBlocks} />;
+    }
+  };
+
+  return (
+    <div
+      className={classNames(
+        className,
+        "my-4 flex flex-col overflow-hidden rounded-[28px] bg-slate-50 sm:my-0",
+        Array.isArray(contentBlocks) || Array.isArray(assistantBlocks)
+          ? "col-span-12 lg:col-span-10 lg:col-start-2 xl:col-span-8 xl:col-start-3"
+          : "col-span-12 md:col-span-8 md:col-start-3 lg:col-span-6 lg:col-start-4 xl:col-span-4 xl:col-start-5"
+      )}
+    >
+      <div className="flex grow basis-0 flex-col gap-8 p-8 md:flex-row">
+        {renderContentBlocks()}
+      </div>
+      <div
+        className={classNames(
+          "flex grow basis-0 flex-col gap-4 border border-slate-800/10 bg-gradient-to-br p-8 pt-6",
+          SolutionSectionColor[color]
+        )}
+      >
+        <div className="text-sm uppercase text-slate-800/60">
+          {Array.isArray(assistantBlocks)
+            ? "The assistants for the job:"
+            : "The assistant for the job:"}
+        </div>
+        <div className="flex grow basis-0 flex-col justify-start gap-8 md:flex-row">
+          {renderAssistantBlocks()}
+        </div>
+      </div>
+    </div>
+  );
+};
+
+interface SolutionSectionContentBlockProps {
+  title: ReactNode;
+  content: ReactNode;
+}
+SolutionSection.ContentBlock = function SolutionSectionContentBlock({
+  content,
+  title,
+}: SolutionSectionContentBlockProps) {
+  return (
+    <div className={classNames("flex grow basis-0 flex-col gap-3")}>
+      <H5 className="text-slate-900">{title}</H5>
+      <P size="xs" className="max-w-[500px] text-slate-600">
+        {content}
+      </P>
+    </div>
+  );
+};
+
+interface SolutionSectionAssistantBlockProps {
+  name: string;
+  description: string;
+  emoji: string;
+  backgroundColor: string;
+}
+
+SolutionSection.AssistantBlock = function SolutionSectionAssistantBlock({
+  name,
+  description,
+  emoji,
+  backgroundColor,
+}: SolutionSectionAssistantBlockProps) {
```

Comment:
This should use anonymous functions.

Reply:
raises an error when so

Reply:
Will take a look.

Reply:
Ok, tried to fix it but failed.
If you can take a look.